src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -51,44 +51,56 @@
     @Autowired
     UserRepository userRepository;
 
+    @Autowired
+    PublicationService publicationService;
+
     /**
-     * Method to create new subscribers.
-     * @return Returns a map which contains two lists, Errored and Created Subscribers. Created will have object ID set.
+     * Method to create new accounts in azure.
+     *
+     * @param azureAccounts The accounts to be created.
+     * @param issuerEmail The email of the user who created the accounts.
+     *
+     * @return Returns a map which contains two lists, Errored and Created accounts. Created will have object ID set.
      **/
-    public Map<CreationEnum, List<? extends Subscriber>> createSubscribers(List<Subscriber> subscribers) {
+    public Map<CreationEnum, List<? extends AzureAccount>> addAzureAccounts(
+        List<AzureAccount> azureAccounts, String issuerEmail) {
 
-        Map<CreationEnum, List<? extends Subscriber>> processedAccounts = new ConcurrentHashMap<>();
+        Map<CreationEnum, List<? extends AzureAccount>> processedAccounts = new ConcurrentHashMap<>();
 
-        List<Subscriber> createdAccounts = new ArrayList<>();
-        List<ErroredSubscriber> erroredAccounts = new ArrayList<>();
+        List<AzureAccount> createdAzureAccounts = new ArrayList<>();
+        List<ErroredAzureAccount> erroredAccounts = new ArrayList<>();
 
-        for (Subscriber subscriber : subscribers) {
+        for (AzureAccount azureAccount : azureAccounts) {
 
-            Set<ConstraintViolation<Subscriber>> constraintViolationSet = validator.validate(subscriber);
+            Set<ConstraintViolation<AzureAccount>> constraintViolationSet = validator.validate(azureAccount);
             if (!constraintViolationSet.isEmpty()) {
 
-                ErroredSubscriber erroredSubscriber = new ErroredSubscriber(subscriber);
+                ErroredAzureAccount erroredSubscriber = new ErroredAzureAccount(azureAccount);
                 erroredSubscriber.setErrorMessages(constraintViolationSet
-                                                       .stream().map(ConstraintViolation::getMessage)
+                             .stream().map(constraint -> constraint.getPropertyPath() + ": " + constraint.getMessage())
                                                        .collect(Collectors.toList()));
                 erroredAccounts.add(erroredSubscriber);
                 continue;
             }
 
             try {
-                User user = azureUserService.createUser(subscriber);
-                subscriber.setAzureSubscriberId(user.id);
-
-                createdAccounts.add(subscriber);
+                User user = azureUserService.createUser(azureAccount);
+                azureAccount.setAzureAccountId(user.id);
+                createdAzureAccounts.add(azureAccount);
+                log.info(publicationService.sendNotificationEmail(azureAccount.getEmail(),
```

Comment:
youre not differentiating admin vs media

Reply:
I've amended this but would welcome your thoughts on whether the new approach is okay

Reply:
im not seeing where youve fixed this